graveler/graveler.go

```diff
@@ -1008,8 +1008,76 @@ func (g *Graveler) ResetPrefix(ctx context.Context, repositoryID RepositoryID, b
 	return err
 }
 
-func (g *Graveler) Revert(_ context.Context, _ RepositoryID, _ BranchID, _ Ref) (CommitID, error) {
-	panic("implement me")
+// Revert creates a reverse patch to the commit given as 'ref', and applies it as a new commit on the given branch.
+// This is implemented by merging the parent of 'ref' into the branch, with 'ref' as the merge base.
+// Example: consider the following tree: C1 -> C2 -> C3, with the branch pointing at C3.
+// To revert C2, we merge C1 into the branch, with C2 as the merge base.
```

Comment:
I don't see where you handle a request to revert `C0` which isn't in the branch.

Reply:
synced f2f - the behavior is similar to git, and will apply the reverse change regardless if the commit is an ancestor of the branch head.